pyqir/src/values.rs

```diff
@@ -494,7 +512,7 @@ impl AttributeList {
     fn param(&self, py: Python, n: u32) -> AttributeSet {
         AttributeSet {
             function: self.0.clone_ref(py),
-            index: AttributeLoc::Param(n),
+            index: n + 1,
```

Comment:
Add comment that LLVM offesets param indices by one because 0 is reserved for ret val